petl/io/xlsx.py

```diff
@@ -86,15 +87,54 @@ def __iter__(self):
             pass
 
 
-def toxlsx(tbl, filename, sheet=None, write_header=True):
+def toxlsx(tbl, filename, sheet=None, write_header=True, mode="replace"):
     """
     Write a table to a new Excel .xlsx file.
 
+    N.B., the sheet name is case sensitive.
+
+    The `mode` argument controls how the file and sheet are treated:
+
+    - `replace`: This is the default. It either replaces or adds a
+      named sheet, or if no sheet name is provided, all sheets
+      (overwrites the entire file).
+    - `overwrite`: Always overwrites the file. This produces a file
+      with a single sheet.
+    - `add`: Adds a new sheet. Raises `ValueError` if a named sheet
+      already exists.
+
+    The `sheet` argument can be omitted in all cases. The new sheet
+    will then get a default name.
+
+    If the file does not exist, it will be created, unless `replace`
+    mode is used with a named sheet. In the latter case, the file
+    must exist and be a valid .xlsx file.
+
     """
 
     import openpyxl
-    wb = openpyxl.Workbook(write_only=True)
-    ws = wb.create_sheet(title=sheet)
+
+    if not os.path.exists(filename) or mode == "overwrite" \
```

Comment:
> if not os.path.exists(filename)

Since [v1.6.0](https://github.com/petl-developers/petl/releases/tag/v1.6.0) `petl` allows reading and writing files to remote sources like remote servers or cloud providers when also using [fsspec](https://github.com/intake/filesystem_spec) package.

However, this code will work only with local files.

Until now that is not a problem because the code of `fromxls()`  and `toxls()` are not working with remote sources.

But the PR #506 adds support for remote sources and also conflict with this PR.

Any ideas of what is the best way to conciliate this two changes? :smiley:

Reply:
So if I had sent the PR four weeks earlier, it would be your problem now? Thanks, I think. :-)

I have no clue at all how the remote sources code works, and cannot make any sense of it right now. I'll try to find the time.

Reply:
Hi @chrullrich,

It's absolutely not a problem at all. :smiley:

When using remote sources, `petl` only call open/read/write on python [file objects](https://docs.python.org/3/library/io.html).

So `os.path.exists()` does not work when reading or writing to `HTTP`, `S3`, `SMB` remote sources for example.

Do you think there is any trick for not using  `os.path.exists()` in this functionality?

My current plan is something like:

1. Waith for @alimanfoo  reviewing #502 
2. Find a workaround for  `os.path.exists()`
3. Merge #506 
4. Adapt and merge this PR (#502)